cnf-tests/testsuites/pkg/performanceprofile/performanceprofile.go

```diff
@@ -78,6 +79,50 @@ func FindOrOverridePerformanceProfile(performanceProfileName, machineConfigPoolN
 	return nil
 }
 
+func OverridePerformanceProfile(performanceProfileName, machineConfigPoolName string, newPerformanceProfile *performancev2.PerformanceProfile) error {
+
+	previousPerfProfile, err := FindDefaultPerformanceProfile(performanceProfileName)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		previousPerfProfile = nil
+	}
+
+	mcp := &mcv1.MachineConfigPool{}
+	err = client.Client.Get(context.TODO(), goclient.ObjectKey{Name: machineConfigPoolName}, mcp)
+	if err != nil {
+		return err
+	}
+
+	if previousPerfProfile != nil {
+		OriginalPerformanceProfile = previousPerfProfile.DeepCopy()
```

Comment:
Good point. What would you use for the map key? previous performance profile name?

I guess this is not going to work in any case when tests run in parallel, as the affected nodes could collide and lead to an unexpected cluster state.

Maybe we can get rid of this save/restore mechanism and make tests clear and create PerformanceProfile. WDYT?

Reply:
The problem is, IIRC in some scenarios the performance profile is created upfront to save some running time.

And my concern is not about running tests in parallel, but this not being able to restore the right one. Until now there was only a test that was doing that in an isolated manner, so the global was serving that purpouse only. Now we are using it for two different scenarios, and looking at the API that is not clear.

Reply:
An alternative here is to check if `OriginalPerformanceProfile` is filled already and panic in that case, because that would mean we are re-replacing

Reply:
would it make sense to return the previous perf profile and set the global/map/whatever in the calling site? this will at least make the flow more explicit instead of buried inside this function, which is not great when we mutate the global state.